kernel-plugins/test/foundry/BLSValidator.test.sol

```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import "kernel/src/factory/KernelFactory.sol";
+import "kernel/src/factory/ECDSAKernelFactory.sol";
+
+// // test artifacts
+import "kernel/src/test/TestExecutor.sol";
+import "kernel/src/test/TestERC721.sol";
+// // test utils
+import "forge-std/Test.sol";
+import {ERC4337Utils} from "kernel/test/foundry/ERC4337Utils.sol";
+// // BLS validator
+import {BLSValidator} from "../../src/BLSValidator.sol";
+
+using ERC4337Utils for EntryPoint;
+
+contract BLSValidatorTest is Test {
+    Kernel kernel;
+    KernelFactory factory;
+    ECDSAKernelFactory ecdsaFactory;
+    EntryPoint entryPoint;
+    ECDSAValidator validator;
+    address owner;
+    uint256 ownerKey;
+    address payable beneficiary;
+
+    function setUp() public {
+        (owner, ownerKey) = makeAddrAndKey("owner");
+        entryPoint = new EntryPoint();
+        factory = new KernelFactory(entryPoint);
+
+        validator = new ECDSAValidator();
+        ecdsaFactory = new ECDSAKernelFactory(factory, validator, entryPoint);
+
+        kernel = Kernel(payable(address(ecdsaFactory.createAccount(owner, 0))));
+        vm.deal(address(kernel), 1e30);
+        beneficiary = payable(address(makeAddr("beneficiary")));
+    }
+
+    function test_mode_2_bls() external {
+        BLSValidator blsValidator = new BLSValidator();
+        // Note: With the Kernel wallet validation and execution are seperate
+        // Executors are plugins that add custom functions to Kernel and each function
+        // is tied to a validator.  In this test we will be enabling the blsValidator and
+        // tying the TextExecutor to it.
+        TestExecutor testExecutor = new TestExecutor();
+        UserOperation memory op =
+            entryPoint.fillUserOp(address(kernel), abi.encodeWithSelector(TestExecutor.doNothing.selector));
+
+        // BLS public/private key pair and signatures created using the '@thehubbleproject/bls' library.
+        // Hard coded in the test because I don't know of a way to do this in solidity.
+        uint256[4] memory publicKey = [
+            0x004b1d8408dcc643647e4f32a761853e873cd1da8ffc40f03b00647484b3498a,
+            0x248b9979254108c9cbb2005739dc693f1694b7b2058942114a0ab4aa81723a6d,
+            0x0edd89947147f52246e1dc3092b62c9020afca38d470b3b827e267c66350da93,
+            0x2d3f8d5cc58f02219b73b70a0acf33778335e7b5f5c7bf98a1d596b9270e92bb
+        ];
+
+        bytes memory enableData = abi.encodePacked(
+            publicKey,
+            uint48(0),
+            TestExecutor.doNothing.selector,
+            uint48(0),
+            uint48(0),
+            uint32(16)
+        );
+
+        bytes32 digest = getTypedDataHash(
```

Comment:
Is `getTypedDataHash` just needed for the ECDSA sig?

Reply:
I need to look into this a little more, I think something like this is needed to make the EIP-712 message but I'm not 100% sure.  Right now, my domain is just 

```
bytes32 public constant BLS_DOMAIN = keccak256("eip4337.bls.domain");
```

which is also not good for EIP-712.  I added a comment at the top of the validator plugging to look into EIP-712.

Reply:
yeah my understanding of EIP 712 isn't too good either. Only have a very high level understanding of it so need to look deeper into it too